src/odemis/driver/autoscript_client.py

```diff
@@ -0,0 +1,2161 @@
+        return self.parent.get_stigmator(self.channel)
+
+    def _setRotation(self, rotation):
+        self.parent.set_scan_rotation(rotation, channel=self.channel)
+        return self.parent.get_scan_rotation(self.channel)
+
+    def _setHorizontalFoV(self, fov):
+        self.parent.set_field_of_view(fov, channel=self.channel)
+        fov = self.parent.get_field_of_view(self.channel)
+        mag = self._hfw_nomag / fov
+        self.magnification._value = mag
+        self.magnification.notify(mag)
+        return fov
+
+    def _onHorizontalFoV(self, fov):
+        self._updateDepthOfField()
+        # the dwell time range is dependent on the magnification/horizontalFoV
+        self._updateDwellTimeRng()
+        if self._has_detector:
+            self._updatePixelSize()
+
+    def _updateDepthOfField(self):
+        fov = self.horizontalFoV.value
+        # Formula was determined by experimentation
+        K = 100  # Magical constant that gives a not too bad depth of field
+        dof = K * (fov / 1024)
+        self.depthOfField._set_value(dof, force_write=True)
+
+    def _updateDwellTimeRng(self):
+        """The dwell time range is dependent on the magnification/horizontalFoV, the range whenever the fov updates."""
+        self.dwellTime._set_range(self.parent.dwell_time_info(self.channel)["range"])
+
+    def prepareForScan(self):
+        """
+        Make sure the beam is unblanked when the blanker is in 'auto' mode before starting to scan.
+        """
+        if self.blanker.value is None:
+            self.parent.unblank_beam(self.channel)
+
+    def finishScan(self):
+        """
+        Make sure the beam is blanked when the blanker is in 'auto' mode at the end of scanning.
+        """
+        if self.blanker.value is None:
+            self.parent.blank_beam(self.channel)
+
+
+
+class Detector(model.Detector):
+    """
+    This is an extension of model.Detector class. It performs the main functionality
+    of the SEM. It sets up a Dataflow and notifies it every time that an SEM image
+    is captured.
+    """
+
+    def __init__(self, name: str, role: str, parent: SEM, channel: str, **kwargs):
+        """
+        :param name (str): name of the Detector
+        :param role (str): role of the Detector
+        :param parent (SEM): parent of the Detector
+        :param channel (str): name of the acquistion channel (electron, ion)
+        :param kwargs: additional keyword arguments
+        """
+        # The acquisition is based on a FSM that roughly looks like this:
+        # Event\State |    Stopped    |   Acquiring    | Receiving data |
+        #    START    | Ready for acq |        .       |       .        |
+        #    DATA     |       .       | Receiving data |       .        |
+        #    STOP     |       .       |     Stopped    |    Stopped     |
+        #    TERM     |     Final     |      Final     |     Final      |
+        self.parent: SEM  # for type hinting
```

Comment:
Not sure of this is needed because in the `Component` class there is a setter and getter

```
    @property
    def parent(self):
        if self._parent:
            return self._parent()
        else:
            return None

    @parent.setter
    def parent(self, p):
        if p:
            assert isinstance(p, Component)
            self._parent = weakref.ref(p)
        else:
            self._parent = None

```